spring-batch-infrastructure/src/main/java/org/springframework/batch/item/file/mapping/BeanWrapperFieldSetMapper.java

```diff
@@ -102,14 +102,25 @@
 
 	private boolean strict = true;
 
+	public BeanWrapperFieldSetMapper(Class<? extends T> type) {
+		setTargetType(type);
```

Comment:
You should never call non-final methods from a constructor - some subclass could override the method and not invoke the super.

Probably not a big deal and unlikely to be a problem for a setter but generally best practice.